src/main/java/org/opensearch/ad/client/AnomalyDetectionNodeClient.java

```diff
@@ -32,4 +35,11 @@ public void searchAnomalyResults(SearchRequest searchRequest, ActionListener<Sea
             listener.onResponse(searchResponse);
         }, listener::onFailure));
     }
+
+    @Override
+    public void getDetectorProfile(ADTaskProfileRequest profileRequest, ActionListener<ADTaskProfileResponse> listener) {
+        this.client.execute(ADTaskProfileAction.INSTANCE, profileRequest, ActionListener.wrap(profileResponse -> {
+            listener.onResponse(profileResponse);
```

Comment:
When using MLClient class transport responses in Flow Framework, we experienced Class Cast Exceptions on these responses because the two different plugins load the same class using different classloaders.

I believe that will be the case here as well.  See the MLClient for implementations that use serialization to generate a new copy of the `profileResponse` by the calling thread.

These are not needed for response types that are pulled from OpenSearch itself.

Reply:
Thanks for the insight! I'll check the ML client. I haven't tested the integration with the skills repo yet. I suppose I didn't run into the issue yet since the 2 transport actions already added are just `SearchRequest`/`SearchResponse` inputs/outputs.

Reply:
I've updated to follow the same pattern as ML client. Also, changed the input to simply be string; the client will generate the request. This prevents any potential classloader issues on request-side, and simplifies the interface and requirements from caller's side, as generating the request is not trivial and requires some AD-specific logic.